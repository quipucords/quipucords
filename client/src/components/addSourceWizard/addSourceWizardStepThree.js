import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Spinner } from 'patternfly-react';
import { connect, reduxSelectors } from '../../redux';

const AddSourceWizardStepThree = ({ add, error, fulfilled, pending, name }) => (
  <React.Fragment>
    {error && (
      <div className="wizard-pf-complete blank-slate-pf">
        <div className="wizard-pf-success-icon">
          <Icon type="pf" name="error-circle-o" />
        </div>
        <h3 className="blank-slate-pf-main-action">Error {add ? 'Creating' : 'Updating'} Source</h3>
        <p className="blank-slate-pf-secondary-action">There are errors on the previous step</p>
      </div>
    )}
    {fulfilled && (
      <div className="wizard-pf-complete blank-slate-pf">
        <div className="wizard-pf-success-icon">
          <Icon type="pf" name="ok" />
        </div>
        <h3 className="blank-slate-pf-main-action">
          <strong>{name}</strong> was {add ? 'created' : 'updated'}.
        </h3>
      </div>
    )}
    {pending && (
      <div className="wizard-pf-process blank-slate-pf">
        <Spinner loading size="lg" className="blank-slate-pf-icon" />
        <h3 className="blank-slate-pf-main-action">{add ? 'Creating' : 'Updating'} Source...</h3>
        <p className="blank-slate-pf-secondary-action">
          Please wait while source <strong>{name}</strong> is being {add ? 'created' : 'updated'}.
        </p>
      </div>
    )}
  </React.Fragment>
);

AddSourceWizardStepThree.propTypes = {
  add: PropTypes.bool,
  error: PropTypes.bool,
  fulfilled: PropTypes.bool,
  pending: PropTypes.bool,
  name: PropTypes.string
};

AddSourceWizardStepThree.defaultProps = {
  add: false,
  error: false,
  fulfilled: false,
  pending: false,
  name: null
};

const makeMapStateToProps = () => {
  const mapSource = reduxSelectors.sources.makeSourceDetail();

  return (state, props) => ({
    ...state.addSourceWizard,
    ...mapSource(state, props)
  });
};

const ConnectedAddSourceWizardStepThree = connect(makeMapStateToProps)(AddSourceWizardStepThree);

export { ConnectedAddSourceWizardStepThree as default, ConnectedAddSourceWizardStepThree, AddSourceWizardStepThree };
