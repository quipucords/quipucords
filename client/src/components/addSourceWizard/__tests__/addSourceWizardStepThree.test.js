import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedAddSourceWizardStepThree, AddSourceWizardStepThree } from '../addSourceWizardStepThree';

describe('AccountWizardStepResults Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component with error', () => {
    const store = generateEmptyStore({
      addSourceWizard: {
        error: true,
        errorMessage: 'Lorem ipsum'
      }
    });
    const component = shallow(<ConnectedAddSourceWizardStepThree />, { context: { store } });

    expect(component).toMatchSnapshot('connected');
  });

  it('should render a wizard results step with error', () => {
    const props = {
      add: false,
      error: true,
      errorMessage: 'lorem ipsum'
    };

    let component = mount(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('error updated');

    props.add = true;
    component = mount(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('error created');
  });

  it('should render a wizard results step with pending', () => {
    const props = {
      add: false,
      pending: true,
      name: 'Dolor'
    };

    let component = mount(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('pending updated');

    props.add = true;
    component = mount(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('pending created');
  });

  it('should render a wizard results step with fulfilled', () => {
    const props = {
      add: false,
      fulfilled: true,
      name: 'Dolor'
    };

    let component = mount(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('fulfilled updated');

    props.add = true;
    component = mount(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('fulfilled created');
  });
});
