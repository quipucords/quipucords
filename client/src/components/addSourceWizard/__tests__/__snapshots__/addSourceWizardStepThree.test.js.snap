// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected 1`] = `
<AddSourceWizardStepThree
  add={false}
  credentials={Array []}
  dispatch={[Function]}
  error={true}
  errorMessage="Lorem ipsum"
  fulfilled={false}
  name={null}
  pending={false}
/>
`;

exports[`error created 1`] = `
<AddSourceWizardStepThree
  add={true}
  error={true}
  errorMessage="lorem ipsum"
  fulfilled={false}
  name={null}
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <Icon
        name="error-circle-o"
        type="pf"
      >
        <PatternflyIcon
          className=""
          name="error-circle-o"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-error-circle-o"
          />
        </PatternflyIcon>
      </Icon>
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      Error 
      Creating
       Source
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      There are errors on the previous step
    </p>
  </div>
</AddSourceWizardStepThree>
`;

exports[`error updated 1`] = `
<AddSourceWizardStepThree
  add={false}
  error={true}
  errorMessage="lorem ipsum"
  fulfilled={false}
  name={null}
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <Icon
        name="error-circle-o"
        type="pf"
      >
        <PatternflyIcon
          className=""
          name="error-circle-o"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-error-circle-o"
          />
        </PatternflyIcon>
      </Icon>
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      Error 
      Updating
       Source
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      There are errors on the previous step
    </p>
  </div>
</AddSourceWizardStepThree>
`;

exports[`fulfilled created 1`] = `
<AddSourceWizardStepThree
  add={true}
  error={false}
  fulfilled={true}
  name="Dolor"
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <Icon
        name="ok"
        type="pf"
      >
        <PatternflyIcon
          className=""
          name="ok"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-ok"
          />
        </PatternflyIcon>
      </Icon>
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      <strong>
        Dolor
      </strong>
       was 
      created
      .
    </h3>
  </div>
</AddSourceWizardStepThree>
`;

exports[`fulfilled updated 1`] = `
<AddSourceWizardStepThree
  add={false}
  error={false}
  fulfilled={true}
  name="Dolor"
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <Icon
        name="ok"
        type="pf"
      >
        <PatternflyIcon
          className=""
          name="ok"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-ok"
          />
        </PatternflyIcon>
      </Icon>
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      <strong>
        Dolor
      </strong>
       was 
      updated
      .
    </h3>
  </div>
</AddSourceWizardStepThree>
`;

exports[`pending created 1`] = `
<AddSourceWizardStepThree
  add={true}
  error={false}
  fulfilled={false}
  name="Dolor"
  pending={true}
>
  <div
    className="wizard-pf-process blank-slate-pf"
  >
    <Spinner
      className="blank-slate-pf-icon"
      inline={false}
      inverse={false}
      loading={true}
      size="lg"
    >
      <div
        className="blank-slate-pf-icon spinner spinner-lg"
      />
    </Spinner>
    <h3
      className="blank-slate-pf-main-action"
    >
      Creating
       Source...
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      Please wait while source 
      <strong>
        Dolor
      </strong>
       is being 
      created
      .
    </p>
  </div>
</AddSourceWizardStepThree>
`;

exports[`pending updated 1`] = `
<AddSourceWizardStepThree
  add={false}
  error={false}
  fulfilled={false}
  name="Dolor"
  pending={true}
>
  <div
    className="wizard-pf-process blank-slate-pf"
  >
    <Spinner
      className="blank-slate-pf-icon"
      inline={false}
      inverse={false}
      loading={true}
      size="lg"
    >
      <div
        className="blank-slate-pf-icon spinner spinner-lg"
      />
    </Spinner>
    <h3
      className="blank-slate-pf-main-action"
    >
      Updating
       Source...
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      Please wait while source 
      <strong>
        Dolor
      </strong>
       is being 
      updated
      .
    </p>
  </div>
</AddSourceWizardStepThree>
`;
